### ISSUE TITLE
Deselecting "Use System Default Output Device" causes crash.

### ISSUE THREAD

**Issue by FreyarH** [Yellow-Dog-Man/Resonite-Issues#1704]
### Describe the bug?

When using the new settings menu in Beta 2024.4.15.1407, deselecting "Use System Default Output Device" causes a crash.

### To Reproduce

Start Resonite in Desktop Mode
Let Dashboard Open
Select "Settings"
Select "Audio"
Click "Use System Default Output Device" to disable it.

### Expected behavior

The setting will disable and the button to configure Audio Output Devices will be enabled so you can change output audio devices.

### Screenshots

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/77557994/2572c392-fc6d-42ae-bc53-e6eaf4161dfa)
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/77557994/edea4dde-5d49-4e25-bf84-33a2edf05a32)

### Resonite Version Number

Beta 2024.4.15.1407

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop (Untested w/ Vive Pro Eye)

### Log Files

[X3DBOX - 2024.4.15.1407 - 2024-04-15 21_30_53.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/14986480/X3DBOX.-.2024.4.15.1407.-.2024-04-15.21_30_53.log)
[Player.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/14986491/Player.log)

### Additional Context

There are 14 output devices registered for playback on this system. (10 disabled, one disconnected, nvidia sure does like to count every display device as an audio device...)
There are 6 input devices registered for recording. Deselecting "Use Default Device" for input does not cause this crash.

### Reporters

@Freyar
@marsmaantje
@MaxyGates

**Comment by FreyarH** [comment]
Referenced in 1721, and that may have a point. I have one virtual audio devices that pushes up to 7.1 audio channels.

Could be choking on enumerating that device.
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/77557994/24521c23-922f-4ffa-b6a1-aa6e52105d54)

**Comment by Nytra** [comment]
I'm having a crash which I think is related. I can launch Resonite and deselect "Use System Default Output Device" without crashing, however if I then switch my default device to `OUT 1-4 (BEHRINGER UMC 204HD 192k)` I crash instantly.

The output device at launch which works fine:

`OUT 1-2 (BEHRINGER UMC 204HD 192k), ID: {0.0.0.00000000}.{a69178f9-b62b-4697-a7cc-dcde3ec9239d}, Format: ChannelsAvailable: 2|SampleRate: 48000|Bps: 384000|BlockAlign: 8|BitsPerSample: 32|Encoding: Extensible|SubFormat: 00000001-0000-0010-8000-00aa00389b71|ChannelMask: SpeakerFrontLeft, SpeakerFrontRight, Default: True`

The device which crashes me:

`OUT 1-4 (BEHRINGER UMC 204HD 192k), ID: {0.0.0.00000000}.{56e928aa-10fb-4cd6-b557-18d371f5d642}, Format: ChannelsAvailable: 4|SampleRate: 48000|Bps: 768000|BlockAlign: 16|BitsPerSample: 32|Encoding: Extensible|SubFormat: 00000001-0000-0010-8000-00aa00389b71|ChannelMask: SpeakerFrontLeft, SpeakerFrontRight, SpeakerBackLeft, SpeakerBackRight, Default: False`

Logs:

[DESKTOP-H976HO2 - 2024.4.26.1258 - 2024-04-28 22_36_21.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/15143913/DESKTOP-H976HO2.-.2024.4.26.1258.-.2024-04-28.22_36_21.log)

[Player.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/15143914/Player.log)

**Comment by shiftyscales** [comment]
Yeah, that device meets the suspected criteria of the above report in that it has more than two channels, @Nytra. Thanks for the additional logs, as far as I can tell this issue already has all that it needs to be looked into.

Seeking input from any of @Frooxius, @ProbablePrime, or @Geenz.

**Comment by Cataena** [comment]
Not sure if helpful to add, but it happens every time I turn my bluetooth headphones on and they connect to my PC, since I'm on "use default device" and it auto-switches to them. It's causing pretty frequent crashes for me since my headphones tend to time out every so often and turn themselves off.

**Comment by shiftyscales** [comment]
@FreyarH @marsmaantje @Nytra Can you retest on the latest build? I suspect this may be resolved with the closure of #1739.

Likewise, @Cataena - #1739 sounds like it may have been your issue, so it would be good to check if that issue is still occurring for you.

**Comment by Nytra** [comment]
@shiftyscales I am still crashing when trying to switch to the `OUT 1-4 (BEHRINGER UMC 204HD 192k)` device.

New log files:

[DESKTOP-H976HO2 - 2024.5.24.202 - 2024-05-24 22_48_09.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/15440331/DESKTOP-H976HO2.-.2024.5.24.202.-.2024-05-24.22_48_09.log)

[Player.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/15440334/Player.log)

**Comment by FreyarH** [comment]
@shiftyscales Problem persists on Beta 2024.5.24.202

[X3DBOX - 2024.5.24.202 - 2024-05-24 16_43_09.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/15440616/X3DBOX.-.2024.5.24.202.-.2024-05-24.16_43_09.log)

**Comment by Frooxius** [comment]
From the logs, this looks unfortunately like a bug in CSCore library, probably due to it tripping over certain audio devices. 

We would like to switch away from it completely, since it's not really maintained anymore, possibly to NAudio. I'm not sure if I'd want to commit to making fixes for it right now because of that.

If anybody wants to poke around and potentially open PR with a fix, here's out fork: https://github.com/Yellow-Dog-Man/cscore

**Comment by Nytra** [comment]
This crash still happens when using the Awwdio prerelease Beta 2025.4.8.119

**Comment by Nytra** [comment]
Can confirm I am no longer crashing when selecting the `OUT 1-4 (BEHRINGER UMC 204HD 192k)` device as default

**Comment by Frooxius** [comment]
Sweet thank you! I'm closing this one!
